src/interface/blas2_interface.hpp

```diff
@@ -728,10 +731,21 @@ typename sb_handle_t::event_t _tpmv_impl(sb_handle_t& sb_handle, index_t _N,
       tpmv, static_cast<index_t>(local_range_y * local_range_x),
       roundUp<index_t>(local_range_y * vector_size,
                        local_range_y * local_range_x),
-      static_cast<index_t>(local_range_x * (loc_mem_leading_dim + 2)));
+      static_cast<index_t>(local_range_x * (loc_mem_leading_dim + 2)),
+      _dependencies);
 
   auto assignOp = make_op<Assign>(vx, vres);
-  return concatenate_vectors(tpmvEvent, sb_handle.execute(assignOp));
+  auto ret =
+      concatenate_vectors(tpmvEvent, sb_handle.execute(assignOp, tpmvEvent));
+
+#ifdef DEFAULT_CPU
```

Comment:
Thanks for your patch, @muhammad-tanvir-1211.

It looks good to me. As discuss earlier, we should deallocate the temporary arrays also for the other backends in order to avoid memory leaks.

Reply:
Yes, otherwise we get memory leak. You can create a issue track in dpcpp for this error. Also you if profiling is enabled you can query the queue and do wait on dependencies here and deallocate the memory, otherwise do what you are doing. And remove the default_CPU as it should be for all the cases.

Reply:
Hi @mehdi-goli and @pgorlani,
I am going to address this in another PR as this change is going to effect multiple operators. Thanks.

Reply:
This change has been addressed in https://github.com/codeplaysoftware/sycl-blas/pull/436